src/odbc_connection.cpp

```diff
@@ -1274,147 +1322,104 @@ class CallProcedureAsyncWorker : public ODBCAsyncWorker {
-        false,
-        &alloc_error
+        data->hstmt,
+        data->parameters,
+        data->parameterCount
       );
-      if (alloc_error)
-      {
-        SetError("[odbc] Error allocating or reallocating memory when fetching data. No ODBC error information available.\0");
-        return;
-      }
+
       if (!SQL_SUCCEEDED(return_code)) {
         this->errors = GetODBCErrors(SQL_HANDLE_STMT, data->hstmt);
-        SetError("[odbc] Error retrieving information about procedures\0");
+        SetError("[odbc] Error getting information about procedure parameters\0");
         return;
       }
 
-      if (data->storedRows.size() == 0) {
-        char errorString[255];
-        #ifndef UNICODE
-        sprintf(errorString, "[odbc] CallProcedureAsyncWorker::Execute: Stored procedure '%s' doesn't exist", combinedProcedureName);
-        #else
-        sprintf(errorString, "[odbc] CallProcedureAsyncWorker::Execute: Stored procedure '%S' doesn't exist", combinedProcedureName);
-        #endif
-        SetError(errorString);
-        return;
-      }
-
-      data->deleteColumns(); // delete data in columns for next result set
+      SQLHDESC hdesc;
 
       return_code =
-      set_fetch_size
-      (
-        data,
-        1
+      SQLGetStmtAttr
+      (  
+        data->hstmt,
+        SQL_ATTR_IMP_PARAM_DESC,
+        &hdesc,
+        SQL_IS_POINTER,
+        0
       );
 
-      return_code = 
-      SQLProcedureColumns
-      (
-        data->hstmt,     // StatementHandle
-        data->catalog,   // CatalogName
-        SQL_NTS,         // NameLengh1
-        data->schema,    // SchemaName
-        SQL_NTS,         // NameLength2
-        data->procedure, // ProcName
-        SQL_NTS,         // NameLength3
-        NULL,            // ColumnName
-        SQL_NTS          // NameLength4
-      );
-      if (!SQL_SUCCEEDED(return_code)) {
-        this->errors = GetODBCErrors(SQL_HANDLE_STMT, data->hstmt);
-        SetError("[odbc] Error retrieving information about the columns in the procedure\0");
-        return;
-      }
+      int i;
+      SQLSMALLINT buffer;
 
-      return_code = prepare_for_fetch(data);
-      return_code =
-      fetch_all_and_store
-      (
-        data,
-        false,
-        &alloc_error
-      );
-      if (alloc_error)
+      for (i = 0; i < data->parameterCount; i++)
       {
-        SetError("[odbc] Error allocating or reallocating memory when fetching data. No ODBC error information available.\0");
-        return;
-      }
-      if (!SQL_SUCCEEDED(return_code)) {
-        this->errors = GetODBCErrors(SQL_HANDLE_STMT, data->hstmt);
-        SetError("[odbc] Error retrieving the result set containing information about the columns in the procedure\0");
-        return;
-      }
+        return_code =
+          SQLGetDescField
+          (
+            hdesc,
+            i + 1,
+            SQL_DESC_PARAMETER_TYPE,
+            &buffer,
+            0,
+            0
+          );
```

Comment:
> For an IPD, the field is set to SQL_PARAM_INPUT by default if the IPD is not automatically populated by the driver (the SQL_ATTR_ENABLE_AUTO_IPD statement attribute is SQL_FALSE). An application should set this field in the IPD for parameters that are not input parameters.

I don't know that we can rely on this for all drivers, unfortunately. Looks like it's supposed to be off by default, so it seems our driver is behaving contrary to spec: https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/automatic-population-of-the-ipd